packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -1402,6 +1401,42 @@ describe('CozyClient', () => {
         receiveQueryResult('allTodos', fakeResponse)
       )
     })
+
+    const setupOnError = () => {
+      const error = new TypeError('Failed to fetch')
+      requestHandler.mockReturnValueOnce(Promise.reject(error))
+      const onError = jest.fn().mockReturnValueOnce(() => true)
+      return onError
+    }
+
+    it('should call onError callback when an error is catched. It is passed as argument', async () => {
+      const onError = setupOnError()
+
+      try {
+        await client.query(query, { as: 'allTodos', onError })
+      } catch (e) {} // eslint-disable-line no-empty
```

Comment:
Do we really need this try / catch block? Since we should not throw the error because we have have onError callback, I guess this block is not mandatory

Reply:
You are right, it's not mandatory and it's logical